src/network/protocol.rs

```diff
@@ -300,10 +300,20 @@ pub struct FixedIp {
     pub subnet_id: String,
 }
 
+/// A port's IP address.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct AllowedAddressPair {
+    pub ip_address: net::IpAddr,
+    #[serde(skip_serializing_if = "String::is_empty")]
+    pub mac_address: String,
+}
+
 /// A port.
 #[derive(Debug, Clone, Deserialize, Serialize)]
 pub struct Port {
     pub admin_state_up: bool,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
```

Comment:
If it can be missing on deserialization, please add `default`.

Reply:
I don't think it can - https://docs.openstack.org/api-ref/network/v2/?expanded=show-port-details-detail,list-ports-detail#show-port-details doesn't list it as optional.

Reply:
Actually, it says it's an extension plugin, so I'll do this to be safe.